tests/migration/sle12_online_migration/zypper_migration.pm

```diff
@@ -56,8 +57,17 @@ sub run {
             send_key "y";
             send_key "ret";
         }
-        elsif ($out =~ $zypper_migration_error
-            || $out =~ $zypper_migration_conflict
+        elsif ($out =~ $zypper_migration_error) {
+            $zypper_migration_error_cnt += 1;
+            die 'Migration failed with zypper error' if $zypper_migration_error_cnt > 3;
+            record_soft_failure("[$zypper_migration_error_cnt/3] zypper migration error");
+            diag 'Sleep for awhile and retry...';
```

Comment:
`record_soft_failure` usually records workaround for bug.
Better to use `record_info('short msg', 'detailed msg', result => 'softfail')` here.
Other LGTM

Reply:
I think you are right, PR updated.